crates/iota-types/src/base_types.rs

```diff
@@ -317,6 +318,15 @@ impl MoveObjectType {
         }
     }
 
+    pub fn is_coin_manager(&self) -> bool {
+        match &self.0 {
+            MoveObjectType_::GasCoin | MoveObjectType_::StakedIota | MoveObjectType_::Coin(_) => {
+                false
+            }
+            MoveObjectType_::Other(s) => CoinManager::is_coin_manager(s),
+        }
```

Comment:
Since we only seek for the `MoveObjectType_::Other` variant we could replace the logic using this one line

```suggestion
        matches!(&self.0, MoveObjectType_::Other(s) if CoinManager::is_coin_manager(s))
```

Reply:
Done